libbpfgo.go

```diff
@@ -121,6 +122,22 @@ int bpf_prog_detach_cgroup_legacy(
 	return syscall(__NR_bpf, BPF_PROG_DETACH, &attr, sizeof(attr));
 }
 
+struct bpf_link * bpf_prog_attach_iter(struct bpf_program *prog, __u32 map_fd)
+{
+	if (map_fd <= 0) {
```

Comment:
`map_fd` is unsigned, so it is always positive or 0. A negative comparison is misleading.